plaso/parsers/safari_cookies.py

```diff
@@ -43,104 +43,127 @@ def __init__(self):
     self.url = None
 
 
-class BinaryCookieParser(interface.FileObjectParser):
+class BinaryCookieParser(dtfabric_parser.DtFabricBaseParser):
   """Parser for Safari Binary Cookie files."""
 
   NAME = 'binary_cookies'
   DESCRIPTION = 'Parser for Safari Binary Cookie files.'
 
-  _FILE_HEADER = construct.Struct(
-      'file_header',
-      construct.Bytes('signature', 4),
-      construct.UBInt32('number_of_pages'),
-      construct.Array(
-          lambda ctx: ctx.number_of_pages,
-          construct.UBInt32('page_sizes')))
-
-  _COOKIE_RECORD = construct.Struct(
-      'cookie_record',
-      construct.ULInt32('size'),
-      construct.Bytes('unknown_1', 4),
-      construct.ULInt32('flags'),
-      construct.Bytes('unknown_2', 4),
-      construct.ULInt32('url_offset'),
-      construct.ULInt32('name_offset'),
-      construct.ULInt32('path_offset'),
-      construct.ULInt32('value_offset'),
-      construct.Bytes('end_of_cookie', 8),
-      construct.LFloat64('expiration_date'),
-      construct.LFloat64('creation_date'))
-
-  _PAGE_HEADER = construct.Struct(
-      'page_header',
-      construct.Bytes('header', 4),
-      construct.ULInt32('number_of_records'),
-      construct.Array(
-          lambda ctx: ctx.number_of_records,
-          construct.ULInt32('offsets')))
+  _DEFINITION_FILE = 'safari_cookies.yaml'
+
+  _SIGNATURE = b'cook'
```

Comment:
Add the null byte here as well, to match the format specification

Reply:
no the signature is 4 bytes the 0-byte in the format spec is there to reduce false positives